### PR TITLE
NIFI-1800: Tie Controller Services to Process Groups

### PR DESCRIPTION
**Pull request by markap14**

### CODE REVIEW

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/FlowController.java [hunk 1]
```diff
@@ -262,7 +266,7 @@
     private final RemoteSiteListener externalSiteListener;
     private final AtomicReference<CounterRepository> counterRepositoryRef;
     private final AtomicBoolean initialized = new AtomicBoolean(false);
-    private final ControllerServiceProvider controllerServiceProvider;
+    private final StandardControllerServiceProvider controllerServiceProvider;
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/FlowController.java]: Should we need to be referring to the implementation type here?

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/reporting/AbstractReportingTaskNode.java [hunk 2]
```diff
@@ -254,4 +254,9 @@ public void verifyCanStart(final Set<ControllerServiceNode> ignoredReferences) {
     public String toString() {
         return "ReportingTask[id=" + getIdentifier() + ", name=" + getName() + "]";
     }
+
+    @Override
+    protected String getProcessGroupIdentifier() {
+        return "root";
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/reporting/AbstractReportingTaskNode.java]: ReportingTasks should still be scoped globally (by the Controller if you will).

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/StandardProcessorNode.java [hunk 3]
```diff
@@ -1421,4 +1420,10 @@ public void run() {
             callback.postMonitor();
         }
     }
+
+    @Override
+    protected String getProcessGroupIdentifier() {
+        final ProcessGroup group = getProcessGroup();
+        return group == null ? null : group.getIdentifier();
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/StandardProcessorNode.java]: What does returning null here imply?

**Comment by mcgilman** [comment]
@markap14 looks like a great start so far! 

Just starting to dig in on testing and it appears the listing of Controller Services for a particular group isn't working. I'm still getting back every service instead of those defined in the current group (and presumably any ancestor groups).

Also noticed the displayName isn't being populated for PropertyDescriptor in components that are not in the root Proces Group. It appears the StandardControllerServiceProvider (which is invoked to get the displayName) only considers the root Process Group.

When attempting to run against a 0.x flow.xml.gz where the ControllerServices are present outside of the rootGroup, I receive the following error at start up

```
java.lang.IllegalArgumentException:  is not a valid version for Flow serialization. Should be in format <number>.<number>
	at org.apache.nifi.controller.serialization.FlowEncodingVersion.parse(FlowEncodingVersion.java:90) ~[nifi-framework-core-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at org.apache.nifi.controller.serialization.FlowEncodingVersion.parse(FlowEncodingVersion.java:71) ~[nifi-framework-core-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
```

**Comment by markap14** [comment]
@mcgilman sorry - it appears that I messed up with I pushed the changes to the branch, before I created the PR. What you looked at was an incomplete version from a few days ago. Can you please review the changes?

Thanks!
-Mark

**Comment by mcgilman** [comment]
After updating all the issues I was seeing have been addressed. Merged into master. Thanks!

**Pull request closed without merging**